implementation/src/main/java/io/smallrye/mutiny/groups/UniAndGroup2.java

```diff
@@ -71,4 +71,29 @@ private <O> Uni<O> combine(BiFunction<T1, T2, O> combinator) {
         return super.combinedWith(function);
     }
 
+    /**
+     * Creates the resulting {@link Uni}. The items are combined using the given combinator function,
+     * and the resulting {@code Uni<Uni<O>>} is flattened.
+     *
+     * @param combinator the combinator function, must not be {@code null}
+     * @param <O> the type of item
+     * @return the resulting {@link Uni}. The items are combined into a {@link Tuple2 Tuple2&lt;T1, T2&gt;}.
+     */
+    @CheckReturnValue
+    public <O> Uni<O> combinedWithUni(BiFunction<T1, T2, Uni<O>> combinator) {
```

Comment:
I found "with" a bit weird. I don't have any better.
`combinedIntoUni` can also be misleading.

Reply:
Yes!

Reply:
@lordvlad what do you think of `combineAsUni`?

Reply:
😇

Reply:
wait, why should we change the tense? `combine_D_With` vs `combine__AsUni?

Reply:
Shall I introduce an alias and mark `combinedWith` as deprecated?

Reply:
But then again .... `as...` implies that the argument should be a type, whereas the argument is a function...

I think I'd rather stick with `combinedWithUni`. Or maybe `combinedWithFlat` (or `combinedWithFlattened`?).

Reply:
There is no way to vote in github comments, is there? :D

Reply:
> But then again .... `as...` implies that the argument should be a type, whereas the argument is a function...
> 
> I think I'd rather stick with `combinedWithUni`. Or maybe `combinedWithFlat` (or `combinedWithFlattened`?).

That's fair.

How about `.with((...) -> value)` and `.withUni((...) -> uni)`? /cc @cescoffier

Reply:
I like `with` and `withUni`. I could introduce an alias and mark `combinedWith` as deprecated. WDYT?

Reply:
Yes! That looks good!

Reply:
ready for another review @jponge @cescoffier